lasagne/layers/pool.py

```diff
@@ -260,11 +260,11 @@ def get_output_for(self, input, **kwargs):
         num_feature_maps_out = num_feature_maps // self.pool_size
 
         pool_shape = ()
-        for k in range(self.axis):
-            pool_shape += (input.shape[k],)
-        pool_shape += (num_feature_maps_out, self.pool_size)
-        for k in range(self.axis + 1, input.ndim):
-            pool_shape += (input.shape[k],)
+        for axis, length in enumerate(input.shape):
+            if axis == self.axis:
+                pool_shape += (num_feature_maps_out, self.pool_size)
+            else:
+                pool_shape += (length,)
```

Comment:
Looks cleaner. But why isn't this just `pool_shape = input.shape[:self.axis] + (num_feature_maps_out, self.pool_size) + input.shape[self.axis+1:]`?